src/jdk.incubator.vector/share/classes/jdk/incubator/vector/VectorShuffle.java

```diff
@@ -314,6 +318,36 @@ public static <E> VectorShuffle<E> fromArray(VectorSpecies<E> species, int[] sou
         return vsp.shuffleFromArray(sourceIndexes, offset);
     }
 
+    /**
+     * Creates a shuffle for a given species from
+     * an {@link MemorySegment} starting at a byte offset.
+     *
+     * <p> For each shuffle lane, where {@code N} is the shuffle lane
+     * index, the integer in-memory at index {@code offset + N * 4} is validated
```

Comment:
We need to say a little more about how the integer is obtained from the segment e.g. 

> ..., the integer read from the segment at index ... using an {@link ValueLayout#JAVA_INT_UNALIGNED unaligned `int`} layout is validated ...